NAMESPACE

```diff
@@ -30,6 +30,7 @@ export(rleidv)
 export(rowid)
 export(rowidv)
 export(as.xts.data.table)
+if(getRversion() >= "3.6.0") S3method(xts::as.xts, data.table)
```

Comment:
and how that refers to the fact we do not import/depend xts? maybe we can/should add `requireNamespace` there too?

Reply:
but then xts wouldn't be optional? wouldn't it have to move from suggests to imports?

Reply:
I'm wondering what as.xts.data.table even does and why leaving it to as.xts.data.frame won't do. Maybe we can remove it.

Reply:
I don't know how R parse NAMESPACE in regards to `::` and dependencies defined in `DESCRIPTION`. Doing `if(getRversion() >= "3.6.0" && requireNamespace("xts", silently=TRUE))` wouldn't make `xts` mandatory, `requireNamespace` returns FALSE when no package found. Maybe `requireNamespace` is not allowed in NAMESPACE, don't know that. I don't see any warnings on any build. So seems legit.

Reply:
The fact is that every use of Suggested dependencies was always escaped with `requireNamespace` and here is the first place AFAIK where it isn't escaped.

Reply:
I escape I mean not try to access xts namespace if it is not available. I haven't heard about that new mechanism, if so the check for R 3.6 handles it already.